components/site/src/sass/serde.rs

```diff
@@ -0,0 +1,477 @@
+use config::Config;
+use errors::{Context, Result};
+
+use serde::{Serialize, Serializer};
+
+use std::fmt::Display;
+
+/// write the site's `Config` struct out as a SCSS map literal.  this
+/// first converts the `Config` to a `SassConfig` to only retain and
+/// serialize the keys in the config document that we want to end up
+/// in the resulting SCSS map.
+pub(crate) fn serialize_config(config: &Config) -> Result<String> {
+    let sass_config = config.sass_config();
+    let mut ser = SassMapSerializer::default();
+
+    serde::Serialize::serialize(&sass_config, &mut ser)
+        .context("failed to serialize Zola config document")?;
+
+    Ok(ser.output)
+}
+
+/// custom serde `Serializer` that serializes a structure as an SCSS
+/// map literal.  the primary difference between an SCSS map literal
+/// and JSON is that SCSS uses parentheses like `("key": "value")`
+/// instead of `{"key": "value"}` to express maps/dictionaries.
+#[derive(Default)]
+struct SassMapSerializer {
+    output: String,
+}
+
+#[derive(Debug)]
+struct SassMapSerializerError(String);
+
+type SassMapSerializerResult = std::result::Result<(), SassMapSerializerError>;
+
+impl serde::ser::Error for SassMapSerializerError {
+    fn custom<T: Display>(msg: T) -> Self {
+        SassMapSerializerError(msg.to_string())
+    }
+}
+
+impl Display for SassMapSerializerError {
+    fn fmt(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+        formatter.write_fmt(format_args!("SassMapSerializerError({})", self.0))
+    }
+}
+
+impl std::error::Error for SassMapSerializerError {}
+
+impl<'a> Serializer for &'a mut SassMapSerializer {
+    type Ok = ();
+    type Error = SassMapSerializerError;
+
+    type SerializeSeq = Self;
+    type SerializeTuple = Self;
```

Comment:
You can use the type here so you don't repeat multiple times the same impl for different kinds like tuple/seq

Reply:
See https://github.com/Keats/tera2/blob/master/src/value/ser.rs#L16 for an example